client/js/screens/Profile/ProfileContainer.js

```diff
@@ -21,8 +35,46 @@ class ProfileContainer extends Component {
   });
 
   render() {
-    return <Profile navigation={this.props.navigation} />;
+    return (
+      <UserContext.Consumer>
+        {({ removeUserIdToken, id }) => {
+          if (!id) {
+            this.props.navigation.navigate("Auth");
+          } else {
+            return (
+              <Query query={profileScreenQuery} variables={{ id }}>
+                {({ loading, error, data }) => {
+                  if (loading)
+                    return (
+                      <View style={{ flex: 1, justifyContent: "center" }}>
+                        <ActivityIndicator size="large" color="#1CC6B1" />
+                      </View>
+                    );
+                  if (error) return <Text>{error}</Text>;
+                  if (data) {
+                    return (
+                      <Profile
+                        navigation={this.props.navigation}
+                        data={data}
+                        logout={removeUserIdToken}
+                        id={id}
+                      />
+                    );
+                  }
+                }}
+              </Query>
+            );
+          }
+        }}
+      </UserContext.Consumer>
+    );
   }
 }
 
 export default ProfileContainer;
+
+ProfileContainer.defaultProps = {
+  removeUserIdToken: null,
+  id: null,
+  data: null
```

Comment:
these are not props. They are function arguments. 

the only prop here is navigation.